docs/DeveloperGuide.md

```diff
@@ -1,13 +1,249 @@
 # Developer Guide
 
+## Table of contents
+
+* [Acknowledgements](#acknowledgements)
+* [Design & implementation](#design--implementation)
+  * [Storage Component](#storage-component)
+    * [Design considerations](#design-considerations)
+  * [Visualization Feature](#visualization-feature-)
+    * [Class diagram](#class-diagram)
+    * [Sequence diagram](#sequence-diagram-)
+  * [Add income/expense feature](#add-incomeexpense-feature)
+    * [Step 1](#step-1)
+    * [Step 2](#step-2)
+    * [Step 3](#step-3)
+    * [Step 4](#step-4)
+    * [Diagrams](#diagrams)
+  * [Budget Feature](#budget-feature)
+    * [Set and update budget](#set-and-update-budget)
+    * [Delete budget](#delete-budget)
+    * [Reset budget](#reset-budget)
+    * [View budget](#view-budget)
+  * [Product Scope](#product-scope)
+    * [Target user profile](#target-user-profile)
+    * [Value proposition](#value-proposition)
+  * [User Stories](#user-stories)
+  * [Non-Functional Requirements](#non-functional-requirements)
+  * [Glossary](#glossary)
+  * [Instructions for manual testing](#instructions-for-manual-testing)
+
 ## Acknowledgements
 
+**Xchart (A Simple Charting Library for Java)**
+- author: KNOWN
+- source: [https://knowm.org/open-source/xchart/](https://knowm.org/open-source/xchart/)
+
+**JSON Simple (simple Java toolkit for encoding and decoding JSON)**
+- author: Yidong Fang (Google Code)
+- source: [https://code.google.com/archive/p/json-simple/](https://code.google.com/archive/p/json-simple/)
+
+**Apache Common Langs 3**
+- author: Apache Commons
+- source: [https://commons.apache.org/proper/commons-lang/](https://commons.apache.org/proper/commons-lang/)
+
+**Alpha Vantage Stock Market API**
+- author: Alpha Vantage
+- source: [https://www.alphavantage.co/](https://www.alphavantage.co/)
+
+**Gson Java library**
+- author: Google
+- source: [https://github.com/google/gson](https://github.com/google/gson)
+
+**Financial Modeling Prep Stock API**
+- author: Financial Modeling Prep
+- source: https://site.financialmodelingprep.com/
+
+**round() method in Cashflow.java**
+  - author: mhadidg
+  - source: [https://stackoverflow.com/questions/2808535/round-a-double-to-2-decimal-places]()
+
+**capitalize() method in Cashflow.java**
+  - author: Nick Bolton
+  - source: [https://stackoverflow.com/questions/1892765/how-to-capitalize-the-first-character-of-each-word-in-a-string]()
+    
+**DG adapted from**
+
+* [Addressbook-level3](https://github.com/se-edu/addressbook-level3)
+
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
 ## Design & implementation
```

Comment:
Perhaps it would be good to have a Architecture Diagram showcasing your overall design and mechanism of FinancialPlanner. This would give the Developer their needed context should they want to further implement other features to integrate into your program